includes/pipeline_page/output.php

```diff
@@ -0,0 +1,99 @@
+<?php
+// Build the HTML for a pipeline documentation page.
+// Imported by public_html/pipeline.php - pulls a markdown file from GitHub and renders.
+
+# Details for parsing markdown file, fetched from Github
+# Build the remote file path
+# Special case - root docs is allow
+
+# General docs page
+if($path_parts[1] == 'output'){
+    if(substr($_SERVER['REQUEST_URI'], -3) == '.md'){
+        # Clean up URL by removing .md
+        header('Location: '.substr($_SERVER['REQUEST_URI'], 0, -3));
+        exit;
+    }
+    $filename = implode('/', array_slice($path_parts, 1)).'.md';
+}
+# Must be the readme
+else {
+    $filename = 'README.md';
+    $md_trim_before = '# Introduction';
+}
+
+# Build the local and remote file paths based on whether we have a release or not
+if($pipeline->last_release !== 0){
+  $git_branch = 'master';
+  $local_fn_base = dirname(dirname(dirname(__FILE__)))."/markdown/pipelines/".$pipeline->name."/".$pipeline->last_release."/";
+} else {
+  $git_branch = 'dev';
+  $local_fn_base = dirname(dirname(dirname(__FILE__)))."/markdown/pipelines/".$pipeline->name."/".$pipeline->pushed_at."/";
+}
+$local_md_fn = $local_fn_base.$filename;
+$markdown_fn = 'https://raw.githubusercontent.com/'.$pipeline->full_name.'/'.$git_branch.'/docs/'.$filename;
+
+# Check if we have a local copy of the markdown file and fetch if not
+if(file_exists($local_md_fn)){
+  $markdown_fn = $local_md_fn;
+} else {
+  # Build directories if needed
+  if (!is_dir(dirname($local_md_fn))) {
+    mkdir(dirname($local_md_fn), 0777, true);
+  }
+  $md_contents = file_get_contents($markdown_fn);
+  if($md_contents){
+    file_put_contents($local_md_fn, $md_contents);
+    $markdown_fn = $local_md_fn;
+  } else {
+    # Edge case: No releases, but dev branch doesn't exist - use master instead
```

Comment:
Nice, but this should never happen 😅  We have checks for this at https://nf-co.re/pipeline_health

I'm a little tempted to let this fail...